user/tests/wiring/api/system.cpp

```diff
@@ -81,18 +81,21 @@ test(system_freememory) {
 }
 
 test(system_waitfor) {
+#if PLATFORM_ID!=10
```

Comment:
Rather than hard-code on platform IDs, you can include hal_platform.h and test against HAL_PLATFORM_WIFI, or, since this is at the Wiring level, we also have Wiring_WiFi conditional define from the wiring API. This will future-proof the code for any other non-wifi devices.